integration/app/todo.state.ts

```diff
@@ -30,8 +30,12 @@ export class TodoState implements NgxsOnInit {
     return state.filter(s => s.indexOf('panda') > -1);
   }
 
-  ngxsOnInit(ctx: StateContext<string[]>) {
-    ctx.setState(['NgxsOnInit todo']);
+  ngxsOnInit({ getState, setState }: StateContext<string[]>) {
+    const state: string[] = getState();
+    const payload = 'NgxsOnInit todo';
+    if (!state.includes(payload)) {
+      setState([...state, payload]);
+    }
```

Comment:
this is not necessary, this is a bug that the previous author introduced

Reply:
Does the ngxsOnInit get run after the state has been restored using HMR?
If so, then they need some way of determining if they need to do what they have intended to do on a fresh state.

Reply:
No, I can return these changes in the app.component file, but this is a bug of the previous author, who added a test for ssr

there is a hmr life cycle, and I believe that there is no need to complicate and write more functions to determine what has changed

Reply:
I think we need to think through the potential use cases of how someone might be using the `ngxsOnInit` and how it applies to HMR. For example they could be using it to set up some initial state through some dispatched events. In this case they may want to prevent this from happening if loading as a result of HMR.
On the other hand the user could be starting a timer to refresh a security token in a specific interval. They would want this code to execute, HMR or not.

The decisions above need to be made if the `ngxsOnInit` is run AFTER the state is restored by HMR. But, if the `ngxsOnInit` is run BEFORE the state is restored by HMR then it becomes a lot simpler. Hopefully my question make sa bit more sense given this context...

> Does the ngxsOnInit get run after the state has been restored using HMR?
> If so, then they need some way of determining if they need to do what they have intended to do on a fresh state.

So I guess what I am trying to say is that we have one of 2 options: 
1. Ensure that the HMR state is restored only after the `ngxsOnInit` methods are run
2. Otherwise we need to provide a way for the user to query if HMR has restored state

Reply:
And this is the reason I stopped working on my own implementation a few days ago.
The most interesting case is:
- Access a route that lazy load a state and make a change in this state.
- Move to other route and make a change in the files to activate the HMR.
- Wait for the HMR to finish reloading.
- Enter the same lazy loaded route.